### PR TITLE
Issue #1788: Views: Allow exporting/importing a view code from the UI.

### PR DESCRIPTION
**Pull request by ghost**
Fixes backdrop/backdrop-issues#1788

### CODE REVIEW

**Comment by sentaidigital** [comment]
Travis CI fails on one test looking for "X-Backdrop-Cache: HIT" and only when testing on PHP 7.  We may have an issue with the test framework.

**Comment by klonos** [comment]
Thanx @sentaidigital! Works as expected, but am I the only one annoyed by the fact that the link does not open in a new tab? Can we please have it like that?

**Comment by sentaidigital** [comment]
@quicksketch 

Travis CI Job [4254](https://travis-ci.org/backdrop/backdrop/builds/127643890) failed one test.
Travis CI Job [4256](https://travis-ci.org/backdrop/backdrop/builds/127654405) passed.

The two builds have the exact same code.  The only difference between the two is a git commit message.  Load?  Race condition? I don't know, but it's something to watch for.

**Comment by sentaidigital** [comment]
@klonos I don't think the the D7 version opened in a new window, either.  I tend to open the dropdown, right-click Export and select "open in new tab" to get the export without losing the editor.

**Comment by klonos** [comment]
@sentaidigital if you develop the habit of entering `Fixes [link-to-issue]` as the PR issue summary, then the PR is automatically linked to the issue. Makes it easier to find that way. So, in this PR, `Fixes https://github.com/backdrop/backdrop-issues/issues/1788` Thanx :wink: 

Fair enough about the right-click. Especially if that is how it was done in D7. I have filed https://github.com/backdrop/backdrop-issues/issues/1833 for making these links open a modal anyways.

**Comment by sentaidigital** [comment]
I thought @Gormartsen or @quicksketch set up a github hook to link when the title starts "Issue #xxxx".  I forgot when first entering this PR, and editing the PR's title didn't trigger the link.

Which is the hook looking for, text or title?  I'll try to remember to include it in the future.

**Comment by klonos** [comment]
Yeah, I think that both should work, but it seems not. I usually do both title and summary text just to be sure.

> ...and editing the PR's title didn't trigger the link.

Try entering `Fixes https://github.com/backdrop/backdrop-issues/issues/1788` as the PR summary (1st comment) to see if that does the trick. I would do it for you, but I only have permissions to edit other people's summaries in the issue queue (for adding information and tasks in to-do lists etc). I do not have permissions to edit other people's PR summaries though.

**Comment by sentaidigital** [comment]
Sweet!

**Comment by sentaidigital** [comment]
Well, it gets a reference.

**Comment by klonos** [comment]
backdrop-pr-referenced

:wink:

File: core/modules/views_ui/views_ui.admin.inc [hunk 1]
```diff
@@ -1254,6 +1258,10 @@ function views_ui_render_display_top($view, $display_id) {
         'title' => t('clone view'),
         'href' => "admin/structure/views/view/$view->name/clone",
       ),
+      'export' => array(
+        'title' => t('export view'),
+        'href' => 'admin/config/development/configuration/single/export/Views/views.view.' . $view->name,
+      ),
```
Comment (quicksketch) [inline core/modules/views_ui/views_ui.admin.inc]: Could you add a `module_exists('config')` and permissions check (`synchronize configuration`) here?

**Comment by sentaidigital** [comment]
Added `module_exists()` and permission checks.

**Comment by quicksketch** [comment]
Thanks @sentaidigital! The random failing is definitely annoying and not related to your changes. We have an issue for this at https://github.com/backdrop/backdrop-issues/issues/1478.

**Comment by quicksketch** [comment]
Merged into 1.x for the 1.4.0 release thanks @sentaidigital!

**Comment by sentaidigital** [comment]
Ah, I was going to add a test for this over the weekend, but I can create a new issue for that.

**Comment by quicksketch** [comment]
That would still be greatly appreciated! You can piggy-back adding the tests to the existing issue at https://github.com/backdrop/backdrop-issues/issues/1788 if you like.

**Pull request merged**